api/src/main/java/net/kyori/adventure/chat/ChatType.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.chat;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import net.kyori.examination.Examinable;
+import net.kyori.examination.ExaminableProperty;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A type of chat.
+ *
+ * @since 4.12.0
+ * @sinceMinecraft 1.19
+ */
+public interface ChatType extends Examinable, Keyed {
+  /**
+   * A chat message from a player.
+   *
+   * @since 4.12.0
+   * @sinceMinecraft 1.19
+   */
+  ChatType CHAT = new ChatTypeImpl(Key.key("chat"));
```

Comment:
```suggestion
  Keyed CHAT = Key.key("chat"));
```

and so on (after #779)

and have `ChatType` be an empty interface for now -- I fully expect things to change for 1.19.2/whatever mojang calls it, so I'd like to avoid stating too much about this api structure as-is